### PR TITLE
Bug 1872919: Add routes for OCS OLM pages with full resource references

### PR DESCRIPTION
**Pull request by bipuladh**

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 1]
```diff
@@ -134,6 +134,19 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+  {
+    type: 'Page/Route',
+    properties: {
+      exact: true,
+      path: `/k8s/ns/:ns/${referenceForModel(
+        ClusterServiceVersionModel,
+      )}/:appName/${apiObjectRef}/~new`,
+      loader: () =>
+        import('./components/ocs-install/install-page' /* webpackChunkName: "install-page" */).then(
+          (m) => m.default,
```
Comment (gnehapk) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:146]: Shouldn't the above `'Page/Route` should be updated with these details? Why another object is required?

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:146]: We need to support both.

**Review by gnehapk — APPROVED** [review]

File: frontend/packages/noobaa-storage-plugin/src/plugin.ts [hunk 2]
```diff
@@ -92,6 +123,25 @@ const plugin: Plugin<ConsumedExtensions> = [
       required: [NOOBAA_FLAG],
     },
   },
+  {
+    type: 'Page/Route',
+    properties: {
+      exact: true,
+      path: [
+        `/k8s/ns/:ns/${referenceForModel(ClusterServiceVersionModel)}/:appName/${referenceForModel(
```
Comment (afreen23) [inline frontend/packages/noobaa-storage-plugin/src/plugin.ts:131]: Would be better if you could extract `referenceForModel(ClusterServiceVersionModel)` once in  a variable and use everywhere.

**Review by afreen23 — CHANGES_REQUESTED** [review]
We need to do this chnage for LSO as well, for `localvolumeset` and `autodetect`.

File: frontend/packages/noobaa-storage-plugin/src/plugin.ts (hunk 2 above)
Comment (bipuladh) [reply frontend/packages/noobaa-storage-plugin/src/plugin.ts:131]: Keeping it this way IMO keeps it more readable. It conveys that we are using the GVK name of the CSVModel.

**Pull request merged**